Paternity Test.py

```diff
@@ -0,0 +1,378 @@
+#
+# This sort of thing can be done on a local UNIX/UNIX-like machine using shell scripts, but that isn't ideal for certain BYOD scenarios. We need a way to do it programmatically (as there might be hundreds of files) and on the Terra platform itself (as the files might be coming from Windows, or the user doesn't know how to run a shell script), hence the motivation for this notebook's creation.
+#
+# ## Assumptions
+# 1) You have a psuedo-folder in your Terra data section that contains BYOD data in the form of files
+# 2) All of the files you want in your data table are in the same psuedo-folder, and are not in sub-psuedo-folders
+# 3) You've uploaded all your files already -- Terra data tables are not dynamic, so if you want to add more later, you'll have to re-run this notebook
+# 4) You are not trying to overwrite a data table you have already created. If you re-run this notebook and set TABLE_NAME to something that already exists as a Terra data table, the old table will NOT to overwritten. You will need to either pick a new name or delete the old table first.
+#
+# You do NOT have to have run File Finder before this notebook. The difference between this and File Finder is that File Finder will create a row for every file in your psuedo-folder, while Paternity Test will create a row for every parent file.
+#
+# Files that lack the parent file extension and are not children will NOT be added to the data table.
+
+# Version history:
+#
+# | v | date | author | notes |
+# | --- | --- | --- | --- |
+# | 0.9 | April 6th 2020 | Ash | initial |
+# | 1.0 | April 13th 2020 | Ash, Lon |   fixed multiple children and no parent usecases, implented Lon's no-pandas code, implemented Brian and Lon's other suggestions, better intro|
+# | 1.1 | April 14th 2020 | Ash |   made Pandas code significantly faster|
+# | 1.2 | April 27th 2020 | Ash |   fixed NameError exception notice, rearranged for clarity and consistency across notebooks|
+#
+
+# # Setup
+
+# ## Imports
+
+# In[ ]:
+
+
+import io
+import os
+import firecloud.api as fapi
+import google.cloud.storage
+
+# Only used in multiple children use case
+import pandas as pd
+import logging
+
+
+# ## User-set variables
+
+# In[ ]:
+
+
+# Don't forgot the quotation marks!
+SUBDIRECTORY="/thousands/"
+TABLE_NAME="paternity_test_test" #Do not include spaces or weird characters
+PARENT_FILETYPE="cram"
+
+# If your filenames are like this, please set INCLUDE_PARENT_EXTENSION to True:
+# NWD119844.CRAM
+# NWD119844.CRAM.CRAI
+
+# If you filenames are like this, please set INCLUDE_PARENT_EXTENSION to False:
+# NWD119844.CRAM
+# NWD119844.CRAI
```

Comment:
Originally I did this in order to allow the program to make links between the files because the pandas use case doesn't have the user specify the child extension. But since the pandas case extracts the extensions anyway, I think I can get rid of it. Thanks!